internal/log_analysis/log_processor/pantherlog/meta.go

```
@@ -0,0 +1,338 @@
+		return errors.Errorf(`duplicate field name %q`, field.Name)
+	}
+	if _, duplicateFieldNameJSON := fieldsByName[field.Name]; duplicateFieldNameJSON {
+		return errors.Errorf(`duplicate JSON field name %q`, field.Name)
+	}
+	registeredFields[id] = field.StructField()
+	fieldNamesJSON[id] = field.NameJSON
+	// Store both the JSON name and the go field name
+	fieldsByName[field.Name] = id
+	fieldsByName[field.NameJSON] = id
+	return nil
+}
+
+// DefaultFields returns the default panther 'any' fields.
+// It creates a new copy so that outside packages cannot affect the defaults.
+func DefaultFields() []FieldID {
+	return []FieldID{
+		FieldIPAddress,
+		FieldDomainName,
+		FieldSHA256Hash,
+		FieldSHA1Hash,
+		FieldMD5Hash,
+		FieldTraceID,
+	}
+}
+
+var defaultMetaFields = DefaultFields()
+
+// FieldMeta describes a panther 'any' field.
+type FieldMeta struct {
+	Name        string
+	NameJSON    string
+	Description string
+}
+
+func (m *FieldMeta) StructField() reflect.StructField {
+	tag := fmt.Sprintf(`json:"%s,omitempty" description:"%s"`, m.NameJSON, m.Description)
+	return reflect.StructField{
+		Name: m.Name,
+		Tag:  reflect.StructTag(tag),
+		Type: typStringSlice,
+	}
+}
+
+func MustBuildEventSchema(event interface{}, fields ...FieldID) interface{} {
+	schema, err := BuildEventSchema(event, fields...)
+	if err != nil {
+		panic(err)
+	}
+	return schema
+}
+
+func BuildEventSchema(event interface{}, fields ...FieldID) (interface{}, error) {
+	typ := reflect.TypeOf(event)
+	for typ.Kind() == reflect.Ptr {
+		typ = typ.Elem()
+	}
+
+	eventType, err := BuildEventTypeSchema(typ, fields...)
+	if err != nil {
+		return nil, err
+	}
+	tmp := reflect.New(eventType)
+	return tmp.Interface(), nil
+}
+
+func BuildEventTypeSchema(eventType reflect.Type, extras ...FieldID) (reflect.Type, error) {
+	fields, err := extendStructFields(nil, eventType)
+	if err != nil {
+		return nil, err
+	}
+	fields, _ = extendStructFields(fields, reflect.TypeOf(CoreFields{}))
```

hmmm, why ignore the error? Seems you should be returning it

---

Because the only error is if the `reflect.Type` is not a struct and we know it is.

---

What about adding a good comment explaining why we ignore the error or handle it or panic? It is a bit confusing for a new reader, considering as well that we handle the error above.